addons/l10n_it_edi/models/account_edi_format.py

```diff
@@ -514,178 +514,187 @@ def _import_fattura_pa(self, tree, invoice):
                 move_type = "in_invoice"
                 _logger.info('Document type not managed: %s. Invoice type is set by default.', document_type)
 
-            # Setup the context for the Invoice Form
-            invoice_ctx = invoice.with_company(company) \
-                                 .with_context(default_move_type=move_type)
+            vals = {
+                'company_id': company.id,
+                'move_type': move_type,
+            }
+
+            # Collect extra info from the XML that may be used by submodules to further put information on the invoice lines
+            extra_info, message_to_log = self._l10n_it_edi_get_extra_info(company, document_type, body_tree)
 
-            # move could be a single record (editing) or be empty (new).
-            with invoice_ctx._get_edi_creation() as invoice_form:
+            partner = self._l10n_it_get_partner_invoice(tree, company)
+            if partner:
+                vals['partner_id'] = partner.id
+            else:
+                message_to_log.append("%s<br/>%s" % (
+                    _("Vendor not found, useful informations from XML file:"),
+                    invoice._compose_info_message(
+                        tree, './/CedentePrestatore')))
 
-                # Collect extra info from the XML that may be used by submodules to further put information on the invoice lines
-                extra_info, message_to_log = self._l10n_it_edi_get_extra_info(company, document_type, body_tree)
+            # Numbering attributed by the transmitter. <1.1.2>
+            elements = tree.xpath('//ProgressivoInvio')
+            if elements:
+                vals['payment_reference'] = elements[0].text
 
-                partner = self._l10n_it_get_partner_invoice(tree, company)
-                if partner:
-                    invoice_form.partner_id = partner
+            elements = body_tree.xpath('.//DatiGeneraliDocumento//Numero')
+            if elements:
+                vals['ref'] = elements[0].text
+
+            # Currency. <2.1.1.2>
+            currency = self.env.ref('base.EUR')
+            elements = body_tree.xpath('.//DatiGeneraliDocumento/Divisa')
+            if elements:
+                currency_str = elements[0].text
+                currency = self.env.ref('base.%s' % currency_str.upper(), raise_if_not_found=False)
+                if currency != self.env.company.currency_id and currency.active:
+                    vals['currency_id'] = currency.id
```

Comment:
Sure, I think that's possible to receive an XML from a US partner for services that specifies USD as currency and uses EUR for the taxes, and has that "descriptive field" somewhere saying that the USD is only used for lines and not for taxes.
I would definately not remove code that handles a corner case just because we think it won't happen.